corehq/middleware.py

```diff
@@ -81,3 +85,14 @@ def process_response(self, request, response):
             duration = datetime.datetime.utcnow() - request._profile_starttime
             profile_logger.info('{} time {}'.format(request.path, duration), extra={'duration': duration})
         return response
+
+
+class TimeoutMiddleware(object):
+
+    def process_request(self, request):
+        now = datetime.datetime.utcnow()
+        last_request = request.session.get('last_request')
+        if last_request is not None and now - string_to_utc_datetime(last_request) > datetime.timedelta(minutes=settings.INACTIVITY_TIMEOUT):
```

Comment:
This if condition could be extracted to a function `if session_expired(request)`